regimes/fr/tax_identity.go

```diff
@@ -52,11 +59,19 @@ func validateVATTaxCode(value interface{}) error {
 		return errors.New("invalid format")
 	}
 
-	// Extract the last nine digits as an integer.
-	siren := str[2:] // extract last nine digits
-	chk := calculateVATCheckDigit(siren)
-	expectStr := str[:2] // compare with first two digits
-	if chk != expectStr {
+	chk := str[:2]   // first 2
+	siren := str[2:] // last 9
+	if taxCodeWithLetters.MatchString(chk) {
```

Comment:
What if the tax code starts with one letter and one number. I'm not sure if it's possible but https://github.com/ltns35/go-vat/blob/main/countries/france.go covers it. I would do the opposite and check if the first two digits are numbers.
```suggestion
	if !taxCodeWithDigits.MatchString(chk) {
```